GNAT tools try to guess the name of the gcc binary by looking at their
own prefix and suffix. Subsequently they're trying to compile code using
the base gcc if argv[0] does't start with "e".
This hack to their name-guessing hack fixes the problem while touching the
least amount of files.

Index: gcc/ada/osint.adb
--- gcc/ada/osint.adb.orig
+++ gcc/ada/osint.adb
@@ -2243,6 +2243,13 @@ package body Osint is
          Start_Of_Suffix := End_Of_Prefix + Prog'Length + 1;
       end if;
 
+      if Nam = "gcc" and then
+         Name_Buffer (Start_Of_Prefix .. End_Of_Prefix) = "" and then
+         Name_Buffer (Start_Of_Suffix .. Name_Len) = ""
+      then
+         return new String'("egcc");
+      end if;
+
       --  Create the new program name
 
       return new String'
